app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,34 @@
+<section class='recipes_show flex_column'>
+   <h2><%=@recipe.name%></h2>
+   <div class='recipe_information'>
+      <p>Preparation time: <%=@recipe.preparation_time%> minutes</p>
+      <div class='public_logo'>
+         <p><%=@recipe.public ? 'Public' : 'Not Public'%></p>
+         <div class="<%=@recipe.public ? 'public' : 'not_public'%>">
+         <span class="<%=@recipe.public ? 'public_switch' : 'not_public_switch'%>"></span>
+         </div>
+      </div>
+      <p>Cooking time: <%=@recipe.cooking_time%> minutes</p>
+   </div>
+   <p><%=@recipe.description%></p>
+   <div class='recipe_buttons'>
+   <%= button_to(@recipe.public ? "Make Private" : "Make Public", toggle_public_recipe_path(@recipe), method: :put, remote: true) %>
+      <% if can? :update, @recipe%>
+         <%=link_to "Generate Shopping List", general_shopping_list_index_path, class: "button" %>
+         <%=link_to "Add ingredient", new_recipe_recipe_food_path(@recipe), class: "button" %>
+      <%end%>
```

Comment:
- [ ] Dear team you did a great job adding the public and private recipe, but the `make private` option should only be available to the owner of the recipe. Currently, I created the public recipe from user one and opened it from user two, I see an option to make the recipe private even though user two is not the owner of the recipe, also clicking the button ends up throwing an error. Please consider addressing this. 

### The issue 

https://github.com/SanaTameem/recipe-app/assets/69077061/047c8bb4-6966-4f02-8dcb-8cf365d0ad90